src/java.desktop/share/classes/javax/swing/SwingUtilities.java

```diff
@@ -2073,6 +2073,20 @@ static int findDisplayedMnemonicIndex(String text, int mnemonic) {
             return -1;
         }
 
+        if ((KeyEvent.getKeyText(mnemonic).equals("F1")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F2")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F3")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F4")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F5")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F6")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F7")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F8")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F9")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F10")) ||
+           (KeyEvent.getKeyText(mnemonic).equals("F11"))) {
+           return -1;
+        }
+
```

Comment:
There are some issues with this implementation IMHO:
- why invoke `KeyEvent.getKeyText(mnemonic)` eleven times? This creates eleven temporary strings. The result is always the same.
- `KeyEvent.getKeyText()` may return localized strings, which would break the fix
- why not simply compare `mnemonic` with ` KeyEvent.VK_F*`?

Following should do the same:
~~~java
if (mnemonic >= KeyEvent.VK_F1 && mnemonic <= KeyEvent.VK_F11) {
    return -1;
}
~~~

But IMO **all lowercase ascii characters** should be excluded because there are more `VK_` keys in this range (e.g. `setMnemonic(VK_NUMPAD1)` would underline the 'a' character, but `Alt+A` does not click the button):

~~~java
if (mnemonic >= 'a' && mnemonic <= 'z') {
    return -1;
}
~~~

Reply:
Yes, calling KeyEvent.getKeyText(mnemonic) so many times is a mistake, which could have been done once.

I was also sceptical about checking for 'a' to 'z' as spec of method says "This method is only designed to handle character values which fall between 'a' and 'z' or 'A' and 'Z'." so 'a', 'z' are valid characters but this method is for displaying Mnemonic index which is not affected and it prevents the action keys underlining lowercase characters.

Thanks for your suggestion which is incorporated.